### PR TITLE
Add limit to looping in banking-stage

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Forward port of da2078d is missing on master branch.

#### Summary of Changes
The code has diverged. So manually imported the commit to master branch.

Fixes #

### CODE REVIEW

**Review by sakridge — DISMISSED** [review]

**Comment by pgarg66** [comment]
@Lichtso, @alessandrod , could you review the changes in the `LoadedPrograms`? I made these manually, since the code on v1.17 is different.

File: svm/src/transaction_processor.rs
Comment (Lichtso) [inline svm/src/transaction_processor.rs:419]: We should return the existing `loaded_programs_for_txs` here instead of a `Default` one. Otherwise it is not initialized with a slot and environment.

Comment (pgarg66) [reply svm/src/transaction_processor.rs:419]: I copied this from here: https://github.com/solana-labs/solana/commit/da2078d4a07cd10dea97def1a54e5d5a71a8bd20#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR5103
Do you think it's a problem, that should be fixed there as well?

Comment (pgarg66) [reply svm/src/transaction_processor.rs:419]: Also, existing `loaded_programs_for_txs` might have some programs that could've gotten loaded in previous iterations of the loop. So it's not guaranteed to be empty. Maybe we should create a new instance with the correct slot and environment, instead of using `default`.

Comment (Lichtso) [reply svm/src/transaction_processor.rs:419]: It would be cleaner to return the existing one in case something later reads these fields. Can be done in a separate PR if this shall stay a clean forward port.

Comment (pgarg66) [reply svm/src/transaction_processor.rs:419]: Separate PR would be better. Also, maybe setting the correct slot and environment is better than using the existing instance (https://github.com/solana-labs/solana/pull/35342/files#r1506511480).

File: program-runtime/src/loaded_programs.rs
Comment (t-nelson) [inline program-runtime/src/loaded_programs.rs:982]:
```suggestion
        let was_occupied = self.assign_program(key, loaded_program);
        self.loading_task_waiter.notify();
        was_occupied
```

as per original

**Review by Lichtso — DISMISSED** [review]

**Review by t-nelson — APPROVED** [review]
lgtm. optional consistency nit

**Comment by pgarg66** [comment]
Thanks for the review. Just committed @t-nelson nit suggestion.
Need just one approval and I'll merge once CI is green.

**Review by t-nelson — APPROVED** [review]

**Pull request merged**
